### PR TITLE
Save MapSettings in Game

### PR DESCRIPTION
**Pull request by pakfront**
This addresses https://github.com/MegaMek/megamek/issues/3731
It moves the main storage of MapSettings from GameManager to Game so it is serialized on game save. It also forces storage of the board paths to use Linux path separators to make diffing and network messaging more consistent. It has been tested with a macOS and PC client and server.

### CODE REVIEW

**Comment by pakfront** [comment]
need to do a little more testing on midgame savegames

**Comment by pakfront** [comment]
midgame saves work.

File: megamek/src/megamek/common/Game.java [hunk 1]
```diff
@@ -3476,4 +3478,19 @@ public void cancelVictory() {
         setVictoryPlayerId(Player.PLAYER_NONE);
         setVictoryTeam(Player.TEAM_NONE);
     }
+
+    public MapSettings getMapSettings() {
+        if (mapSettings == null)
+        {
```
Comment (Windchild292) [inline megamek/src/megamek/common/Game.java:3484]:
```suggestion
        if (mapSettings == null) {
```

File: megamek/src/megamek/common/Game.java [hunk 2]
```diff
@@ -3476,4 +3478,19 @@ public void cancelVictory() {
...
+        }
+        return mapSettings;
+    }
+
+    public void setMapSettings(MapSettings mapSettings) {
+        this.mapSettings = mapSettings;
+        // clean up linux/macOS/Windows paths
+        this.mapSettings.adjustPathSeparator();
+
+    }
```
Comment (Windchild292) [inline megamek/src/megamek/common/Game.java:3495]:
```suggestion
        this.mapSettings.adjustPathSeparator();
    }
```

**Review by Windchild292 — COMMENTED** [review]
Just two spacing issues, otherwise this looks good.

**Review by Windchild292 — APPROVED** [review]

**Pull request merged**